### PR TITLE
SC-1468

### PR DESCRIPTION
**Pull request by anmol2302**
<b>Ticket Ref:</b> SC-1468
1: Added Feed model and interface and its impl classes

### CODE REVIEW

File: actors/user/src/main/java/org/sunbird/user/service/FeedService.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.sunbird.user.service;
+
+import org.sunbird.common.models.response.Response;
+import org.sunbird.dto.SearchDTO;
+import org.sunbird.models.user.Feed;
+
+public interface FeedService {
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: Interfcae we are starting with I and then name , so follow same

File: actors/user/src/main/java/org/sunbird/user/service/FeedService.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+
+    Response save(Feed feed);
+    Response search(SearchDTO searchDTO);
+    void delete(String id);
```
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: delete should be boolean

File: actors/user/src/main/java/org/sunbird/user/service/FeedService.java (hunk 1 above)
Comment (manzarul) [inline actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: add java doc for interface and method.

File: actors/common/src/main/java/org/sunbird/models/Category.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package org.sunbird.models;
+
+import org.sunbird.common.models.util.JsonKey;
+
+public enum  Category {
+
+    MIGRATION_USER(JsonKey.MIGRATION_USER_OBJECT);
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/models/Category.java]: why enum value is MIGRATION_USER_OBJECT.
i am thinking to use "USER_EXTERNAL_ID_VALIDATION" as enum name and value as well.

File: actors/common/src/main/java/org/sunbird/models/user/Feed.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+import java.io.Serializable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author anmolgupta
+ * Table name : user_feed or user_notifcation
```
Comment (manzarul) [inline actors/common/src/main/java/org/sunbird/models/user/Feed.java]: no need to keep table description here.

File: actors/user/src/main/java/org/sunbird/user/service/FeedService.java (hunk 2 above)
Comment (anmol2302) [reply actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: done

File: actors/user/src/main/java/org/sunbird/user/service/FeedService.java (hunk 1 above)
Comment (anmol2302) [reply actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: done

Comment (anmol2302) [reply actors/user/src/main/java/org/sunbird/user/service/FeedService.java]: done

File: actors/common/src/main/java/org/sunbird/models/user/Feed.java (hunk 4 above)
Comment (anmol2302) [reply actors/common/src/main/java/org/sunbird/models/user/Feed.java]: done

**Pull request merged**